bin/process_output.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+
+from pathlib import Path
+from mimetypes import guess_type
+from functools import partial
+import gzip
+import sys
+import argparse
+import pandas as pd
+
+
+def get_open(f):
+    if "gzip" == guess_type(str(f))[1]:
+        return partial(gzip.open)
+    else:
+        return open
+
+def main(argv=None):
+
+    parser = argparse.ArgumentParser(
+        description="Parses a profile_dists distances to create query-reference-format output for the FastMatch pipeline.",
+        epilog="Example: python process_output.py --input distances.tsv --output results.tsv --threshold 10",
+    )
+
+    parser.add_argument(
+        "--input",
+        action="store",
+        dest="input",
+        type=str,
+        help="profile_dists-generated distance matrix",
+        default=None,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--threshold",
+        action="store",
+        dest="threshold",
+        type=int,
+        help="distance threshold to be included in output",
+        default=None,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--output",
+        action="store",
+        dest="output",
+        type=str,
+        help="output in query-reference format",
+        default=None,
+        required=True,
+    )
+
+    args = parser.parse_args(argv)
+
+    input = Path(args.input)
+    output = Path(args.output)
+    threshold = args.threshold
+
+    data = pd.read_csv(input, sep="\t")
+    data = data[data['Distance'] <= threshold]
+    data.to_csv(output, sep="\t", index=False)
```

Comment:
Since one of the acceptance criteria mentions an Excel file (which are easier to open for people on Windows) could you modify this to also write the results in an excel file (there's a `data.to_excel()` function in pandas, though not sure if you still need to have an Excel writer library installed in Python for this to work).

Reply:
Updated in 4c18be30fbc6ee6c2e50fb7465cbd86df9062c3d

It required openpyxl, which was not in the pandas container. I tried the openpyxl container, but it didn't have pandas, so I ended up using the staramr container, since it has Python, pandas, and openpyxl.